diff --git a/glib/gwin32.c b/glib/gwin32.c
index cc6da5f..9c51e2c 100644
--- a/glib/gwin32.c
+++ b/glib/gwin32.c
@@ -269,29 +269,16 @@ g_win32_get_package_installation_directory_of_module (gpointer hmodule)
   if ((p = strrchr (filename, G_DIR_SEPARATOR)) != NULL)
     *p = '\0';
 
-  retval = g_strdup (filename);
-
-  do
-    {
-      p = strrchr (retval, G_DIR_SEPARATOR);
-      if (p == NULL)
-        break;
-
-      *p = '\0';
-
-      if (g_ascii_strcasecmp (p + 1, "bin") == 0 ||
-          g_ascii_strcasecmp (p + 1, "lib") == 0)
-        break;
-    }
-  while (p != NULL);
+  /* Strip off trailing bin/lib */
+  if ((p = strrchr (filename, G_DIR_SEPARATOR)) != NULL)
+  {
+    if (g_ascii_strcasecmp (p + 1, "bin") == 0 ||
+        g_ascii_strcasecmp (p + 1, "lib") == 0)
+       *p = '\0';
+  }
 
-  if (p == NULL)
-    {
-      g_free (retval);
-      retval = filename;
-    }
-  else
-    g_free (filename);
+  retval = g_strdup (filename);
+  g_free (filename);
 
 #ifdef G_WITH_CYGWIN
   /* In Cygwin we need to have POSIX paths */
