diff --git a/win32/glibpc.py b/win32/glibpc.py
index b04dc24..222d991 100644
--- a/win32/glibpc.py
+++ b/win32/glibpc.py
@@ -26,7 +26,7 @@ def main(argv):
               	          'glib-mkenums': 'perl ${exec_prefix}/bin/glib-mkenums',
               	          'gobject-query': '${exec_prefix}/bin/gobject-query',
               	          '@PCRE_REQUIRES@': '',
-              	          '@INTLLIBS@': '-lintl',
+              	          '@INTLLIBS@': '-llibintl',
               	          '@G_LIBS_EXTRA@': '',
               	          '@PCRE_LIBS@': '',
               	          '@ICONV_LIBS@': '-liconv',
