diff --git a/config.h.win32.in b/config.h.win32.in
index 7bb9a13..8b4458e 100644
--- a/config.h.win32.in
+++ b/config.h.win32.in
@@ -196,10 +196,8 @@
 /* Define to 1 if you have the `hasmntopt' function. */
 /* #undef HAVE_HASMNTOPT */
 
-/* Target the Windows XP API */
-#ifndef _WIN32_WINNT
-#define _WIN32_WINNT 0x0501
-#endif
+/* Target the Windows Vista API */
+#define _WIN32_WINNT 0x0600
 
 #if _WIN32_WINNT >= 0x0600
 /* if_indextoname() and if_nametoindex() are available on Windows Vista/Server
